### PR TITLE
Refactor use object tilespeed

### PR DESCRIPTION
**Pull request by Alberth289346**
Refactor humanoid use-object tile-speed into a separate function and cleanup.

Complete patch is a move with modifications, not quite readable. Patch however takes many small steps so reading those may be preferred.

### CODE REVIEW

**Review by lewri — COMMENTED** [review]
Looks neater, and we aren't making multiple bombardments of `setTilePositionSpeed` now at least :)

File: CorsixTH/Lua/humanoid_actions/use_object.lua [hunk 1]
```diff
@@ -112,6 +112,33 @@ local function action_use_next_phase(action, phase)
...
+
+  -- Decide the animation tile.
+  local tx, ty
+  if obj_orient.use_animate_from_use_position then
+    tx, ty = action.old_tile_x, action.old_tile_y
+  else
+    tx, ty = object:getRenderAttachTile()
+  end
+  if humanoid.humanoid_class == "Handyman" and
+    obj_orient.added_handyman_animate_offset_while_in_use then
```
[the remark this answers is not in the document]
Comment (lewri) [reply CorsixTH/Lua/humanoid_actions/use_object.lua:128]: (it's not you, just pointing it out :) )

**Comment by TheCycoONE** [comment]
LGTM. There is a very similar calculation: https://github.com/CorsixTH/CorsixTH/blob/81db3338d89df253e85fc6b7ace9d937092dafaa/CorsixTH/Lua/humanoid_actions/multi_use_object.lua#L174

**Comment by Alberth289346** [comment]
> LGTM. There is a very similar calculation:
> 
> https://github.com/CorsixTH/CorsixTH/blob/81db3338d89df253e85fc6b7ace9d937092dafaa/CorsixTH/Lua/humanoid_actions/multi_use_object.lua#L174

That looks like an older version of the code, without all the added special cases for positioning animations by data in the footprint.

**Pull request merged**